3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb

```diff
@@ -102,8 +102,18 @@ def initialize(debug, enrich, shred, elasticsearch, s3distcp, config, enrichment
 
         # Configure Elasticity with your AWS credentials
         Elasticity.configure do |c|
-          c.access_key = config[:aws][:access_key_id]
-          c.secret_key = config[:aws][:secret_access_key]
+          aws_session_token = nil
+ +        if config[:aws][:access_key_id] == 'iam' and config[:aws][:secret_access_key] == 'iam'
+            #federated Identity Management
+ +          credentials_from_role = Aws::InstanceProfileCredentials.new.credentials
```

Comment:
Don't forget to add the aws-sdk gem to your Gemfile and Gemfile.lock, same as https://github.com/snowplow/snowplow/commit/c77b8f71671ccf600088899673e0efcd8321ea8e.